islandora_premis.module

```diff
@@ -88,6 +88,9 @@ function islandora_premis_xml_download($output) {
  *   otherwise.
  */
 function islandora_premis_access_callback($object) {
+  if (!user_access('view premis metadata') && !user_access('download premis metadata')) {
+    return FALSE;
```

Comment:
This could probably be jumbled into the big return below without negating I think.